apps/managedidentity/managedidentity_test.go

```diff
@@ -48,95 +80,129 @@ func getSuccessfulResponse(resource string) ([]byte, error) {
 }
 
 func makeResponseWithErrorData(err string, desc string) ([]byte, error) {
-	responseBody := ErrorRespone{
+	responseBody := ErrorResponse{
 		Err:  err,
 		Desc: desc,
 	}
 	jsonResponse, e := json.Marshal(responseBody)
 	return jsonResponse, e
 }
 
-type resourceTestData struct {
-	source   Source
-	endpoint string
-	resource string
-	miType   ID
+func createMockFile(path string, size int64) {
+	dir := filepath.Dir(path)
+	if err := os.MkdirAll(dir, 0755); err != nil {
+		panic(err)
+	}
+
+	f, err := os.Create(path)
+	if err != nil {
+		panic(err)
+	}
+
+	if size > 0 {
+		if err := f.Truncate(size); err != nil {
+			panic(err)
+		}
+	}
+	f.Close()
 }
 
-type errorTestData struct {
-	code          int
-	err           string
-	desc          string
-	correlationid string
+func getMockFilePath() (string, error) {
```

Comment:
Looks like the real file path, on Linux anyway. You could refactor the implementation to make using arbitrary paths in tests simple, [for example](https://github.com/Azure/azure-sdk-for-go/blob/60b1b7ce748104aec8519bc4a8450b04baafc0ce/sdk/azidentity/managed_identity_credential_test.go#L160-L166)

Reply:
Updated with what I think your saying